identity/src/main/java/com/android/identity/credential/PendingAuthenticationKey.kt

```diff
@@ -0,0 +1,227 @@
+                    "is intended to be replaced does not exist")
+            return null
+        }
+
+
+    private lateinit var credential: Credential
+
+    internal var replacementForAlias: String? = null
+
+    /**
+     * Deletes the pending authentication key.
+     */
+    fun delete() {
+        secureArea.deleteKey(alias)
+        credential.removePendingAuthenticationKey(this)
+    }
+
+    /**
+     * Certifies the pending authentication key.
+     *
+     * This will convert this [PendingAuthenticationKey] into a
+     * [AuthenticationKey] including preserving the application-data
+     * set.
+     *
+     * The [PendingAuthenticationKey] object should no longer be used after calling this.
+     *
+     * @param issuerProvidedAuthenticationData the issuer-provided static authentication data.
+     * @param validFrom the point in time before which the data is not valid.
+     * @param validUntil the point in time after which the data is not valid.
+     * @return a [AuthenticationKey].
+     */
+    fun certify(
+        issuerProvidedAuthenticationData: ByteArray,
+        validFrom: Timestamp,
+        validUntil: Timestamp
+    ): AuthenticationKey {
+        return credential.certifyPendingAuthenticationKey(
+            this,
+            issuerProvidedAuthenticationData,
+            validFrom,
+            validUntil
+        )
+    }
+
+    fun toCbor(): DataItem {
+        val builder = CborBuilder()
+        val mapBuilder: MapBuilder<CborBuilder> = builder.addMap()
+        mapBuilder.put("alias", alias)
+            .put("domain", domain)
+            .put("secureAreaIdentifier", secureArea.identifier)
+        if (replacementForAlias != null) {
+            mapBuilder.put("replacementForAlias", replacementForAlias)
+        }
+        mapBuilder.put("applicationData", privateApplicationData.encodeAsCbor())
+            .put("authenticationKeyCounter", authenticationKeyCounter)
+        return builder.build().get(0)
+    }
+
+    companion object {
+        const val TAG = "PendingAuthenticationKey"
+
+        fun create(
+            alias: String,
+            domain: String,
+            secureArea: SecureArea,
+            createKeySettings: CreateKeySettings,
+            asReplacementFor: AuthenticationKey?,
+            credential: Credential
+        ): PendingAuthenticationKey {
+            val ret = PendingAuthenticationKey()
+            ret.alias = alias
+            ret.domain = domain
+            ret.secureArea = secureArea
+            ret.secureArea.createKey(alias, createKeySettings)
+            if (asReplacementFor != null) {
+                ret.replacementForAlias = asReplacementFor.alias
+            }
+            ret.credential = credential
+            ret.privateApplicationData = SimpleApplicationData { ret.credential.saveCredential() }
+            ret.authenticationKeyCounter = credential.authenticationKeyCounter
+            return ret
+        }
```

Comment:
can use the `run` scope function to simplify property accessors of `PendingAuthenticationKey`

```
        fun create(
            alias: String,
            domain: String,
            secureArea: SecureArea,
            createKeySettings: CreateKeySettings,
            asReplacementFor: AuthenticationKey?,
            credential: Credential
        ) = PendingAuthenticationKey().run {
            this.alias = alias
            this.domain = domain
            this.secureArea = secureArea
            this.secureArea.createKey(alias, createKeySettings)
            replacementForAlias = asReplacementFor?.alias
            this.credential = credential
            privateApplicationData = SimpleApplicationData { credential.saveCredential() }
            authenticationKeyCounter = credential.authenticationKeyCounter
        }
```

Note, for line 185, since `replacementForAlias` can be `null`, we can use `?` to do the checking and return either `null` or the value

Reply:
Took this but there's a missing `this` at the end of the `run` block, I think.